### PR TITLE
Add FindMaxRecursionTest

### PR DESCRIPTION
**Pull request by lukasb1b**
Change so it has its own test and not the main methode

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.

### CODE REVIEW

File: src/main/java/com/thealgorithms/maths/FindMaxRecursion.java [hunk 1]
```diff
@@ -49,6 +30,6 @@ public static int max(int[] array, int low, int high) {
      * @return max value of {@code array}
      */
     public static int max(int[] array) {
-        return array.length == 1 ? array[0] : max(array, 0, array.length);
+        return array.length == 1 ? array[0] : max(array, 0, array.length - 1);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/FindMaxRecursion.java:33]:
```suggestion
        return max(array, 0, array.length - 1);
```

File: src/main/java/com/thealgorithms/maths/FindMaxRecursion.java [hunk 2]
```diff
@@ -49,6 +30,6 @@ public static int max(int[] array, int low, int high) {
      * @return max value of {@code array}
      */
     public static int max(int[] array) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/FindMaxRecursion.java:32]:
```suggestion
    public static int max(final int[] array) {
```

Please also mark the arguments of the _other_ `max` method as `final`.

File: src/test/java/com/thealgorithms/maths/FindMaxRecursionTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        Assertions.assertEquals(expected, FindMaxRecursion.max(input));
+    }
+
+    private static Stream<Arguments> inputStream() {
+        return Stream.of(Arguments.of(10, new int[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), Arguments.of(5, new int[] {5, 5, 5, 5, 5}), Arguments.of(0, new int[] {-1, 0}), Arguments.of(-1, new int[] {-10, -9, -8, -7, -6, -5, -4, -3, -2, -1}), Arguments.of(9, new int[] {3, -2, 3, 9, -4, -4, 8}));
+    }
+
+    @Test
+    public void testFindMaxThrowsExceptionForEmptyInput() {
+        assertThrows(IllegalArgumentException.class, () -> FindMax.findMax(new int[] {}));
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/FindMaxRecursionTest.java:26]:
```suggestion
        assertThrows(IllegalArgumentException.class, () -> FindMaxRecursion.max(new int[] {}));
```

File: src/test/java/com/thealgorithms/maths/FindMaxRecursionTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public class FindMaxRecursionTest {
+
+    @ParameterizedTest
+    @MethodSource("inputStream")
+    void numberTests(int expected, int[] input) {
+        Assertions.assertEquals(expected, FindMaxRecursion.max(input));
+    }
+
+    private static Stream<Arguments> inputStream() {
+        return Stream.of(Arguments.of(10, new int[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), Arguments.of(5, new int[] {5, 5, 5, 5, 5}), Arguments.of(0, new int[] {-1, 0}), Arguments.of(-1, new int[] {-10, -9, -8, -7, -6, -5, -4, -3, -2, -1}), Arguments.of(9, new int[] {3, -2, 3, 9, -4, -4, 8}));
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/FindMaxRecursionTest.java:21]:
```suggestion
        return Stream.of(Arguments.of(10, new int[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), Arguments.of(5, new int[] {5, 5, 5, 5, 5}), Arguments.of(0, new int[] {-1, 0}), Arguments.of(-1, new int[] {-10, -9, -8, -7, -6, -5, -4, -3, -2, -1}), Arguments.of(9, new int[] {3, -2, 3, 9, -4, -4, 8}), Arguments.of(3, new int[] {3}));
```

Feel free to remove one of the test cases with _sorted_ array - there are four of them.

**Review by vil02 — CHANGES_REQUESTED** [review]
Please make the `FindMaxRecursion` a _proper utility class_ (i.e. marked as `final` with `private` default constructor).

Idea for future project: make all of the `FindMax` (and `FindMin`) like functions more _uniform_ (i.e. same method names, same tests. etc.)

File: src/main/java/com/thealgorithms/maths/FindMaxRecursion.java (hunk 1 above)
Comment (lukasb1b) [reply src/main/java/com/thealgorithms/maths/FindMaxRecursion.java:33]: I didnt change FindMax Recusion i just saw that error when i had a proper test. Seconf I would rather not use Math.max the performance shhould be equal and so the function works one itself without any other classes.

**Comment by lukasb1b** [comment]
I changed the requests and even the max function

File: src/main/java/com/thealgorithms/maths/FindMaxRecursion.java (hunk 1 above)
Comment (vil02) [reply src/main/java/com/thealgorithms/maths/FindMaxRecursion.java:36]: My suggestion was about not using the `array.length == 1` (this reduces number of branches) and simply call your method `max` with proper arguments.

**Comment by vil02** [comment]
As mentioned earlier please add `private` default constructor.

File: src/main/java/com/thealgorithms/maths/FindMaxRecursion.java [hunk 5]
```diff
@@ -1,26 +1,9 @@
...
-        int[] array = new int[size];
-
-        /* init array with rand numbers */
-        for (int i = 0; i < size; i++) {
-            array[i] = rand.nextInt() % 100;
-        }
-
-        assert max(array) == Arrays.stream(array).max().getAsInt();
-        assert max(array, 0, array.length - 1) == Arrays.stream(array).max().getAsInt();
+    private FindMaxRecursion() {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/FindMaxRecursion.java:5]:
```suggestion
    private FindMaxRecursion() {
```

**Review by vil02 — CHANGES_REQUESTED** [review]

**Review by vil02 — APPROVED** [review]
Awesome!

**Comment by vil02** [comment]
@lukasb1b In order to merge this you need to update your branch (I do not have a permission to commit anything to your branch, so I cannot do it).

**Pull request merged**